pytorch_lightning/trainer/training_loop.py

```diff
@@ -188,10 +189,10 @@ def reset_train_val_dataloaders(self, model) -> None:
         The val dataloader must be initialized before training loop starts, as the training loop
         inspects the val dataloader to determine whether to run the evaluation loop.
         """
-        if self.trainer.train_dataloader is None:
+        if self.trainer.train_dataloader is None or self.trainer.state.status == TrainerStatus.RUNNING:
```

Comment:
The `TrainerStatus` should always be `RUNNING` here. What you added is equivalent to removing the `if` so definitely not the solution